addons/flannel/template/generate.sh

```diff
@@ -10,10 +10,14 @@ function generate() {
     cp -r ./base/* "$dir"
 
     download_yaml
-
+    replace_flannel_image_with_version
     find_images_in_yaml
 }
 
+function replace_flannel_image_with_version() {
+    sed -e "s/flannel:v?(\d+(?:\.\d+)*)/flannel:v$version/g" "$dir/yaml/kube-flannel.yml"
```

Comment:
Why not

```suggestion
    sed -e "s/flannel:v?(\d+(?:\.\d+)(?:\.\d+))/flannel:v$version/g" "$dir/yaml/kube-flannel.yml"
```

That way it only matches x.y.z and not also x.y or x.y.z.a

Reply:
what about the Manifest?